### PR TITLE
Add exception preprocessor

### PR DESCRIPTION
**Pull request by fmeum**
Adds a Kotlin preprocessor for the stack traces of exceptions thrown by the fuzz target to replace the previous string-based C++ severity annotator.

This has two key advantages:
* Special error conditions (such as `VirtualMachineErrors`) can be properly wrapped into a `FuzzerSecurityIssueLow` exception.
* `StackOverflowError` can be properly deduplicated (heuristically). This is necessary as the unprocessed stack traces are very hard to deduplicate, given that their top-most stack frames are essentially random.

### CODE REVIEW

**Comment by fmeum** [comment]
The new `ExampleStackOverflowFuzzer` runs into a segfault on macOS pretty quickly?
```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //examples:ExampleStackOverflowFuzzer
-----------------------------------------------------------------------------
WARNING: Failed to find function "__sanitizer_acquire_crash_state". Reason dlsym(RTLD_DEFAULT, __sanitizer_acquire_crash_state): symbol not found.
WARNING: Failed to find function "__sanitizer_print_stack_trace". Reason dlsym(RTLD_DEFAULT, __sanitizer_print_stack_trace): symbol not found.
INFO: Loaded 8649 no-throw method signatures
INFO: Instrumented com.example.ExampleStackOverflowFuzzer (took 343 ms, size +78%)
INFO: libFuzzer ignores flags that start with '--'
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 2735196724
INFO: Loaded 1 modules   (512 inline 8-bit counters): 512 [0x7faff812c000, 0x7faff812c200), 
INFO: Loaded 1 PC tables (512 PCs): 512 [0x7faff862c000,0x7faff862e000), 
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
/private/var/tmp/_bazel_runner/5b536b8aa56f54cacdf2506d7dc3d463/sandbox/darwin-sandbox/315/execroot/__main__/bazel-out/darwin-opt/bin/examples/ExampleStackOverflowFuzzer.runfiles/__main__/examples/ExampleStackOverflowFuzzer: line 19:  6587 Segmentation fault: 11  driver/jazzer_driver -artifact_prefix=/private/var/tmp/_bazel_runner/5b536b8aa56f54cacdf2506d7dc3d463/sandbox/darwin-sandbox/315/execroot/__main__/bazel-out/darwin-opt/testlogs/examples/ExampleStackOverflowFuzzer/test.outputs/ --reproducer_path=/private/var/tmp/_bazel_runner/5b536b8aa56f54cacdf2506d7dc3d463/sandbox/darwin-sandbox/315/execroot/__main__/bazel-out/darwin-opt/testlogs/examples/ExampleStackOverflowFuzzer/test.outputs -seed=2735196724 --cp=examples/ExampleStackOverflowFuzzer_target_deploy.jar --agent_path=agent/jazzer_agent_deploy.jar -rss_limit_mb=5000
Unexpected exit code: 139
```

@sebastianpoeplau Do you have an idea of what could cause this? It's not too important for now though.

File: driver/fuzz_target_runner.cpp [hunk 1]
```diff
@@ -197,19 +197,20 @@ RunResult FuzzTargetRunner::Run(const uint8_t *data, const std::size_t size) {
     env.DeleteLocalRef(byte_array);
   }
 
-  if (env.ExceptionOccurred()) {
-    jlong dedup_token = computeDedupToken();
+  if (jthrowable exception = env.ExceptionOccurred()) {
+    env.ExceptionClear();
```
Comment (sebastianpoeplau) [inline driver/fuzz_target_runner.cpp:201]: Is it possible that `ExceptionClear` frees the memory of the exception object? That might explain the crash on macOS.

**Review by sebastianpoeplau — APPROVED** [review]
Looks good to me!

File: driver/fuzz_target_runner.cpp (hunk 1 above)
Comment (fmeum) [reply driver/fuzz_target_runner.cpp:204]: That's a good idea, I have pushed a speculative fix and will see what the pipeline says. Unfortunately the JNI doc is completely unclear on this.

Comment (fmeum) [reply driver/fuzz_target_runner.cpp:202]: This segfault does not seem to have anything to do with this PR. Large Java stacks seems to have a history of issues on macOS. I'll mark the test as broken.

**Pull request merged**